File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task197_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.Random;

public class Solution {
    private double radius;
    private double x_center;
    private double y_center;
    private Random random;

    public Solution(double radius, double x_center, double y_center) {
        this.radius = radius;
        this.x_center = x_center;
        this.y_center = y_center;
        this.random = new Random();
    }

    public double[] randPoint() {
        double r = Math.sqrt(random.nextDouble()) * radius;
        double theta = random.nextDouble() * 2 * Math.PI;
        return new double[]{x_center + r * Math.cos(theta), y_center + r * Math.sin(theta)};
    }
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task197_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <random>
#include <cmath>

class Solution {
private:
    double radius;
    double x_center;
    double y_center;
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_real_distribution<double> dis(0.0, 1.0);

public:
    Solution(double radius, double x_center, double y_center) : radius(radius), x_center(x_center), y_center(y_center) {}

    vector<double> randPoint() {
        double r = sqrt(dis(gen)) * radius;
        double theta = dis(gen) * 2 * M_PI;
        return {x_center + r * cos(theta), y_center + r * sin(theta)};
    }
};